Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -0,0 +1,361 @@
+
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(logger, localNuGetSourceDir, localNuGetDownloadDir);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            bool result = apg.Generate(new Roslyn.CommandLine.NuGetReference(Node.A.ToString(), new SemanticVersion("1.0")), "cs", null, outputDir);
+
+            UninstallAllTestNuGetPackages(localNuGetSourceStore);
+
+            // Assert
+            Assert.IsFalse(result, "Generator should fail if a package dependency requires license accept");
+        }
+
+        [TestMethod]
+        public void Generate_PackageNoAccept_MultipleLevelDependencies_SecondLevelSecondDependencyRequiresAccept_Fails()
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+
+            string localNuGetSourceDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetSource");
+            string localNuGetDownloadDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetDownload");
+            IPackageManager localNuGetSourceStore = CreatePackageManager(localNuGetSourceDir);
+            SetupTestGraph(localNuGetSourceStore, Node.E);
+
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(logger, localNuGetSourceDir, localNuGetDownloadDir);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            bool result = apg.Generate(new Roslyn.CommandLine.NuGetReference(Node.A.ToString(), new SemanticVersion("1.0")), "cs", null, outputDir);
+
+            UninstallAllTestNuGetPackages(localNuGetSourceStore);
+
+            // Assert
+            Assert.IsFalse(result, "Generator should fail if a package dependency requires license accept");
+        }
+
+        #region Private methods
+
+        private void UninstallAllTestNuGetPackages(IPackageManager manager)
+        {
+            foreach (Node node in Enum.GetValues(typeof(Node)))
+            {
+                manager.UninstallPackage(node.ToString(), new SemanticVersion("1.0"), true, false);
+            }
+        }
+
+        /// <summary>
+        /// Creates a graph used for testing, with nodes labelled in breadth-first order.
+        /// 
+        /// Visually:
+        /// A
+        /// |\
+        /// B C 
+        /// | |\
+        /// D E F
+        /// </summary>
+        /// <param name="nodesRequireLicense">
+        /// Nodes in the graph that should be packages with the field requireLicenseAccept set to true
+        /// </param>
+        private void SetupTestGraph(IPackageManager mgr, params Node[] nodesRequireLicense)
+        {
+            // leaf nodes
+            CreatePackage(mgr, nodesRequireLicense.Contains(Node.D) ? License.Required : License.NotRequired, Node.D.ToString());
```

Comment:
It depends what you are trying to achieve; readability for the graph creation, or compartmentalisation in the helper method. You could achieve both the creating an intermediate method that includes the license calculation code then calls the existing CreatePackage method.